docs/Storage Rules.md

```diff
@@ -0,0 +1,190 @@
+# Storage Rules
+
+Storage rules are based on Firebase. You can see the [pull request here](https://github.com/nhost/hasura-backend-plus/pull/144).
+
+The rules are governed by a `yaml` file, which defines routes and permissions, with authenticating functions. You can see [the template file here](https://github.com/nhost/hasura-backend-plus/blob/master/custom/storage-rules/rules.yaml)
+
+``` yaml
+functions:
+  isAuthenticated: 'return !!request.auth'
+  isOwner: "return !!request.auth && userId === request.auth['user-id']"
+  validToken: 'return request.query.token === resource.Metadata.token'
+paths:
+  /user/:userId/:
+    list: 'isOwner(userId)'
+  /user/:userId/:fileId:
+    read: 'isOwner(userId) || validToken()'
+    write: 'isOwner(userId)'
+```
+
+In the permission variables cookie returned by the [login]() or [refresh]() endpoints.
+
+This is a URL-encoded string, in the following template:
+
+``` txt
+s:<request.auth>.<checksum>
+```
+
+The request.auth part is a JSON object, in the following layout:
```

Comment:
Maybe mention that the permission variables are the same as the permission variables for Hasura except that we have removed `x-hasura-`.

Reply:
Yep, that is a good point